sloghandler.go

```diff
@@ -127,21 +123,50 @@ func (l *slogHandler) WithGroup(name string) slog.Handler {
 	if l.sink == nil {
 		return l
 	}
+	if name == "" {
+		// slog says to inline empty groups
+		return l
+	}
```

Comment:
This is a bug fix, right?

A separate PR would make it show up in the 1.4 release notes.

Reply:
https://github.com/go-logr/logr/pull/245